File: clientside/zig-movie-app/src/types/Movie.ts
export interface Movie {
  id: number;
  title: string;
  overview: string;
  poster_path: string;
  backdrop_path: string;
  popularity: number;
  vote_average: number;
  vote_count: number;
  release_date: string;
  adult: boolean;
  genre_ids: number[];
  original_language: string;
  original_title: string;
}

export interface MovieDetails extends Movie {
  homepage: string;
  budget: number;
  revenue: number;
  runtime: number;
  status: string;
  tagline: string;
  genres: Genre[];
  production_companies: ProductionCompany[];
}

export interface Genre {
  id: number;
  name: string;
}

export interface ProductionCompany {
  id: number;
  name: string;
  logo_path: string;
  origin_country: string;
}

File: serverside/Models/Movie.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace serverside.Models
{
    public class Movie
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Overview { get; set; }
        [JsonPropertyName("poster_path")]
        public string PosterPath { get; set; }
        [JsonPropertyName("backdrop_path")]
        public string BackdropPath { get; set; }
        public double Popularity { get; set; }
        [JsonPropertyName("vote_average")]
        
        public double VoteAverage { get; set; }
        [JsonPropertyName("vote_count")]
        public int VoteCount { get; set; }
        [JsonPropertyName("release_date")]
        public string ReleaseDate { get; set; }
        public bool Adult { get; set; }
        [JsonPropertyName("genre_ids")]
        public List<int> GenreIds { get; set; }
        [JsonPropertyName("original_language")]
        public string OriginalLanguage { get; set; }
        [JsonPropertyName("original_title")]
        public string OriginalTitle { get; set; }
    }
}